### PR TITLE
Generate ELF shared libraries and allow multi-abi libs in APKs and App bundles

### PR DESCRIPTION
**Pull request by blasten**
Retry #33696

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -397,17 +397,34 @@ class AndroidDevice extends Device {
...
+        androidArch = AndroidArch.arm64_v8a;
+        break;
+      case TargetPlatform.android_x64:
+        androidArch = AndroidArch.x86_64;
+        break;
+      case TargetPlatform.android_x86:
+        androidArch = AndroidArch.x86;
+        break;
+      default:
+      printError('Android platforms are only supported.');
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Nit: indent

**Review by tvolkert — APPROVED** [review]
LGTM

**Review by tvolkert — APPROVED** [review]

**Comment by Hixie** [comment]
If I'm reading the benchmarks correctly, this raised the size of the APK by about 200KB. That seems high. Is that expected?

**Comment by blasten** [comment]
@Hixie https://github.com/flutter/flutter/pull/34250 will bring the size down.

**Comment by Hixie** [comment]
Excellent!

**Pull request merged**